### PR TITLE
Implement block caching

### PR DESCRIPTION
**Pull request by AlexNijjar**
Prevent repeated `get_current_block()` calls by caching the block, only updating it after 12 or more seconds and then passing the cached value to extrinsics.

### CODE REVIEW

File: neuron/neuron/checkpoint.py [hunk 1]
```diff
@@ -87,9 +87,10 @@ def get_submission(
     subtensor: bt.subtensor,
     metagraph: bt.metagraph,
     hotkey: str,
+    block: Optional[int] = None
```
Comment (MsRandom) [inline neuron/neuron/checkpoint.py:90]: Do `int | None` as the type to be consistent.

File: neuron/neuron/checkpoint.py [hunk 2]
```diff
@@ -87,9 +87,10 @@ def get_submission(
...
 ) -> tuple[CheckpointSubmission, int] | None:
     try:
-        metadata = cast(dict[str, Any], get_metadata(subtensor, metagraph.netuid, hotkey))
+        metadata = cast(dict[str, Any], get_metadata(subtensor, metagraph.netuid, hotkey, block))
```
Comment (MsRandom) [inline neuron/neuron/checkpoint.py:93]: Is `block` a positional argument?

Comment (AlexNijjar) [reply neuron/neuron/checkpoint.py:93]: yes

File: validator/validator/validator.py [hunk 3]
```diff
@@ -108,6 +108,9 @@ class Validator:
 
     wandb_run: Run | None
 
+    current_block: int
+    last_block_fetch: datetime = None
```
Comment (MsRandom) [inline validator/validator/validator.py:112]: `datetime | None` so pytype can properly analyze the code

**Review by MsRandom — CHANGES_REQUESTED** [review]

**Review by MsRandom — APPROVED** [review]

**Pull request merged**